crates/isograph_compiler/src/source_files.rs

```diff
@@ -0,0 +1,370 @@
+                self.create_or_update_schema_extension(path)?;
+            }
+            SourceEventKind::Rename((source_path, target_path)) => {
+                if config.schema_extensions.contains(target_path) {
+                    self.create_or_update_schema_extension(target_path)?;
+                } else {
+                    let interned_file_path = intern_file_path(source_path);
+                    self.schema_extensions.remove(&interned_file_path);
+                }
+            }
+            SourceEventKind::Remove(path) => {
+                let interned_file_path = intern_file_path(path);
+                self.schema_extensions.remove(&interned_file_path);
+            }
+        }
+        Ok(())
+    }
+
+    fn handle_update_source_file(
+        &mut self,
+        config: &CompilerConfig,
+        event_kind: &SourceEventKind,
+    ) -> Result<(), BatchCompileError> {
+        match event_kind {
+            SourceEventKind::CreateOrModify(path) => {
+                self.create_or_update_iso_literals(&config.project_root, path)?;
+            }
+            SourceEventKind::Rename((source_path, target_path)) => {
+                let source_file_path = intern_file_path(source_path);
+                if self.contains_iso.remove(&source_file_path).is_some() {
+                    self.create_or_update_iso_literals(&config.project_root, target_path)?
+                }
+            }
+            SourceEventKind::Remove(path) => {
+                let interned_file_path = intern_file_path(path);
+                self.contains_iso.remove(&interned_file_path);
+            }
+        }
+        Ok(())
+    }
+
+    fn handle_update_source_folder(
+        &mut self,
+        config: &CompilerConfig,
+        event_kind: &SourceEventKind,
+    ) -> Result<(), BatchCompileError> {
+        match event_kind {
+            SourceEventKind::CreateOrModify(_) => {}
+            SourceEventKind::Rename((source_path, target_path)) => {
+                let path_string = source_path.to_string_lossy().to_string();
+                self.contains_iso
+                    .retain(|file_path, _| !file_path.to_string().starts_with(&path_string));
+                read_and_parse_iso_literals_from_folder(
+                    &mut self.contains_iso,
+                    target_path,
+                    &config.project_root,
+                )?;
+            }
+            SourceEventKind::Remove(path) => {
+                let path_string = path.to_string_lossy().to_string();
+                self.contains_iso
+                    .retain(|file_path, _| !file_path.to_string().starts_with(&path_string));
+            }
+        }
+        Ok(())
+    }
+
+    fn create_or_update_schema_extension(
+        &mut self,
+        path: &PathBuf,
+    ) -> Result<(), BatchCompileError> {
+        let (file_path, document) = read_and_parse_schema_extensions(path)?;
+        match self.schema_extensions.entry(file_path) {
```

Comment:
since we're not doing anything different in the vacant and occupied cases, this would be cleaner as `self.schema_extensions.insert`. Likewise below for iso literals

Reply:
Indeed, after moving to match file kind first everything became simpler